lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -1016,20 +1009,59 @@ static Value createLinalgPayloadCalculationForElementwiseOp(
       clamp.emitError("unimplemented: runtime optional type");
       return nullptr;
     }
-    auto result = payloadArgs[0];
-    if (!min.getType().isa<Torch::NoneType>()) {
-      auto minPromoted = convertScalarToDtype(b, loc, min, dtype);
-      auto pred = b.create<arith::CmpFOp>(loc, arith::CmpFPredicate::ULT,
-                                          result, minPromoted);
-      result = b.create<arith::SelectOp>(loc, pred, minPromoted, result);
-    }
-    if (!max.getType().isa<Torch::NoneType>()) {
-      auto maxPromoted = convertScalarToDtype(b, loc, max, dtype);
-      auto pred = b.create<arith::CmpFOp>(loc, arith::CmpFPredicate::UGT,
-                                          result, maxPromoted);
-      result = b.create<arith::SelectOp>(loc, pred, maxPromoted, result);
+
+    Type dtype = converter->convertType(clamp.getType())
+                     .cast<RankedTensorType>()
+                     .getElementType();
+    if (dtype.isa<mlir::FloatType>()) {
+      auto result = payloadArgs[0];
+      if (!min.getType().isa<Torch::NoneType>()) {
+        auto minPromoted = convertScalarToDtype(b, loc, min, dtype);
+        auto pred = b.create<arith::CmpFOp>(loc, arith::CmpFPredicate::ULT,
+                                            result, minPromoted);
+        result = b.create<arith::SelectOp>(loc, pred, minPromoted, result);
+      }
+      if (!max.getType().isa<Torch::NoneType>()) {
+        auto maxPromoted = convertScalarToDtype(b, loc, max, dtype);
+        auto pred = b.create<arith::CmpFOp>(loc, arith::CmpFPredicate::UGT,
+                                            result, maxPromoted);
+        result = b.create<arith::SelectOp>(loc, pred, maxPromoted, result);
+      }
+      return result;
+    }
+
+    if (auto intTy = dtype.dyn_cast<mlir::IntegerType>()) {
+      auto result = payloadArgs[0];
+
+      if (!min.getType().isa<Torch::NoneType>()) {
+        auto minPromoted =
+            convertScalarToDtype(b, loc, min, dtype,
+                                 /*srcOriginalDtype=*/std::nullopt,
+                                 /*dstOriginalDtype=*/b.getI64Type());
+        auto pred = b.create<arith::CmpIOp>(loc,
+                                            intTy.isUnsigned()
```

Comment:
In torch-mlir, this should always be signless. See: https://github.com/llvm/torch-mlir/pull/2597. I believe what you want to check is the signedness of `clamp.getType().cast<BaseTensorType>().getDtype()`.

Reply:
I updated the check using the dtype. I have a separate PR in progress that should include a signedness helper.